pkg/block/blocktest/adapter.go

```diff
@@ -26,6 +39,7 @@ func AdapterTest(t *testing.T, adapter block.Adapter, storageNamespace, external
 	t.Run("Adapter_Exists", func(t *testing.T) { testAdapterExists(t, adapter, storageNamespace) })
 	t.Run("Adapter_GetRange", func(t *testing.T) { testAdapterGetRange(t, adapter, storageNamespace) })
 	t.Run("Adapter_Walker", func(t *testing.T) { testAdapterWalker(t, adapter, storageNamespace) })
+	t.Run("Adapter_Namespace", func(t *testing.T) { testAdapterNamespace(t, adapter) })
```

Comment:
Not sure this is the right place as the package level test adapter common capabilities - while storage namespace syntax is per adapter.

Reply:
True, but this way we enforce adding namespace validation test for each block adapter which implements the block adapter tests

Reply:
But it looks like we are adding test code outside the test. The suite doesn't have spec to follow.

Reply:
I agree this is not a spec test, do you have another suggestion as to how we can ensure all block adapters will have a test that verifies the namespace?

Reply:
The thing is that the regex was to help the client enter a valid name for the specific adapter. Each adapter comes with his own regex - if we like to test each regex we can do it in the context of each adapter.
For the spec - which is relevant for all adapters - we need a valid storage namespace like bucket name - if we can read/write to it - it is valid.